pcsx2-qt/Debugger/CpuWidget.cpp

```diff
@@ -1024,16 +1063,53 @@ void CpuWidget::onSearchButtonClicked()
 		m_ui.listSearchResults->clear();
 		const auto& results = workerWatcher->future().result();
 
-		for (const auto& address : results)
-		{
-			QListWidgetItem* item = new QListWidgetItem(QtUtils::FilledQStringFromValue(address, 16));
-			item->setData(256, address);
-			m_ui.listSearchResults->addItem(item);
-		}
+		m_searchResults = results;
+		loadSearchResults();
+		m_ui.btnFilterSearch->setDisabled(m_ui.listSearchResults->count() == 0);
+		
 	});
 
 	m_ui.btnSearch->setDisabled(true);
+	QPushButton* senderButton = qobject_cast<QPushButton*>(sender());
+	bool isFilterSearch = senderButton == m_ui.btnFilterSearch;
+	std::vector<u32> addresses;
+	if (isFilterSearch)
+	{
+		addresses = m_searchResults;
+	}
 	QFuture<std::vector<u32>> workerFuture =
-		QtConcurrent::run(startWorker, &m_cpu, searchType, searchStart, searchEnd, searchValue, searchHex ? 16 : 10);
+		QtConcurrent::run(startWorker, &m_cpu, searchType, addresses, searchStart, searchEnd, searchValue, searchHex ? 16 : 10);
 	workerWatcher->setFuture(workerFuture);
 }
+
+void CpuWidget::onSearchResultsListScroll(u32 value)
+{
+	bool hasResultsToLoad = m_ui.listSearchResults->count() < m_searchResults.size();
```

Comment:
```suggestion
	bool hasResultsToLoad = static_cast<size_t>(m_ui.listSearchResults->count()) < m_searchResults.size();
```
Fixes a sign compare warning